source/process/documentation-UItext-guidelines.rst

```diff
@@ -0,0 +1,193 @@
+
+- Do not use verbs or adjectives as variables.
+- Do not create plurals of variables by adding an *s*.
+- If the variable is a noun, use a qualifier after the variable. For example, say *The {channel_name} channel was created* instead of saying *The {channel_name} was created*.
+
+Messages can be of the following types: notification, confirmation, warning, and error. The following sections contain guidelines that are specific to each of these types.
+
+Notification messages
+~~~~~~~~~~~~~~~~~~~~~~
+
+A notification message is for informing somebody about something that happened. Such messages do not need any user input, and do not prevent a user from continuing to use Mattermost.
+
+- Use either a complete sentence or a sentence phrase. 
+- If using a complete sentence, end it with a period.
+- Examples:
+
+  - *Member added to channel*
+  - *The plug-in was installed.*
+
+Confirmation messages
+~~~~~~~~~~~~~~~~~~~~~~
+
+A confirmation message is for asking somebody to confirm whether the immediately preceding command should be proceeded with. The user cannot use Mattermost until a confirmation is given or denied.
+
+- Use complete sentences.
+- Include a question that has a Yes/No answer.
+- Examples:
+
+  - *Are you sure you want to delete this channel?*
+  - *A plug-in with this ID already exists. Would you like to overwrite it?*
+
+Warning messages
+~~~~~~~~~~~~~~~~
+
+A warning message is for alerting somebody about something that might go wrong. The user can continue using Mattermost unless the warning message needs an explicit user input.
+
+- Use complete sentences.
+- Explain what has happened or can happen, and what can go wrong as a consequence.
+- If the message contains a question, phrase it in such a manner so that it has a Yes/No answer (unless you have specific action buttons for the message).
+
+- Examples:
+
+  - *The Enterprise license expires in 2 days. If you do not renew it, some features will be disabled on licence expiry.*
+  - *If you claim this AD/LDAP account, you will no longer be able to log in with your email. Do you want to continue?*
+
+Error messages
+~~~~~~~~~~~~~~~
+
+An error message is for telling somebody that something went wrong. Errors prevent a user from doing a task or accessing a feature till the error is resolved.
+
+- Use complete sentences.
+- If what went wrong isn't obvious, explain in one sentence.
+- If a solution or workaround isn't obvious, suggest one.
+- Examples:
+
+  - *This message is too long. Shorten it to 120 characters.*
+  - *The passwords do not match.*
+
+
+Guidelines for UI elements
+---------------------------
+Use this table when writing the text for UI elements such as windows, dialog boxes, labels, and prompts.
+
+.. list-table::
+   :widths: 10 30 10 15 35
+   :header-rows: 1
+
+   * - Element
+     - Image 
+     - Capitalization
+     - Phrasing
+     - Examples     
+   * - Menu
+     - |menu.png|
+     - All words
```

Comment:
Are there exceptions? What about words like "as", "of"? For example, "Mark as Unread"?

Reply:
Yes, there are exceptions. Words with 3 letters or fewer, articles, prepositions, and coordinating conjunctions will not be capitalized if for headline style of capitalization. However, since the audience of this style guide would be developers mainly (who might not know about headline style vs. sentence style), I thought I'll simplify it to "All words". Do you think we should change it to "Headline style"?
Since "as" is a subordinating conjunction, it will be capitalzed  :) "Mark As Unread". Nouns, pronouns, adjectives, adverbs, verbs, and subordinating conjunctions are.

Reply:
Yes I think we should make the guidelines as accurate as possible, maybe provide a link to a website or other resource that explains what Headline style is?

Reply:
I added a footnote to the table.